ui/src/services/NodeVolumesUtils.js

```diff
@@ -168,13 +168,26 @@ export const getVolumeListData = createSelector(
     volumeLatencyCurrent,
     volumeCapacityCurrentList,
   ) => {
-    let nodeVolumes = volumes;
-    // filter the volumes by the node name from URL
-    if (nodeName) {
-      nodeVolumes = volumes?.filter(
-        (volume) => volume.spec.nodeName === nodeName,
-      );
+    let nodeVolumes = null;
+
+    if (nodeFilter) {
+      // when there is node filter in URL, check if the node name is valid / exist
+      const node = nodeList?.find((node) => node.name === nodeFilter);
+      if (!node) {
+        return [];
+      } else {
+        // the node name exists, check if there is any volume belong to this node
+        nodeVolumes = volumes?.filter(
+          (volume) => volume.spec.nodeName === nodeFilter,
+        );
+        if (!nodeVolumes) {
+          return [];
+        }
+      }
```

Comment:
That's excessively complex IMO, unless we really want to prevent the case where someone inputs a wrong node name both in the volume spec and in the URL... Otherwise, this would suffice:
```suggestion
      nodeVolumes = volumes?.filter(
        (volume) => volume.spec.nodeName === nodeFilter,
      );
```